### DISCUSSION TITLE
Bar Width with rounded corners

### DISCUSSION THREAD

**Discussion by HenrySteinmetz** [JakeStanger/ironbar#110] [Q&A]
I want to have the bar not take up the entire width of my screen while also having rounded edges (I already got those but I don't have any idea on how to get the bar thinner without using a border)

**Comment by JakeStanger** [comment]
Sorry this slipped past me for a few days, since helpfully all my GitHub emails are now going to spam.

> I want to have the bar not take up the entire width of my screen 

Currently this is not possible, but feel free to open a feature request as I don't think it'd be too difficult to add in, just needs some tweaking in the gtk-layer-shell setup.

I'm confused where the rounded corners come in, is that related to the bar width issue or completely separate?

**Reply by HenrySteinmetz** [reply]
So I thought that I could just use an invisible border to make the bar not take up the whole screen but that doesn't seem to be a good solution. I actually now have a bar that doesn't take the full width (using the anchor to edges option set to false)  but this messes up the music tab
![image](https://user-images.githubusercontent.com/121169024/233779390-a88db839-6d25-48ec-af82-ab8b7446700b.png)

**Reply by JakeStanger** [reply]
Ah, that's a weird consequence of GTK automatically sizing widgets, where it wants to be as small as possible. You can work around this by setting `truncate.mode` and `truncate.length` on the music widget to set it to a fixed size.

**Reply by HenrySteinmetz** [reply]
I tried using the truncate length, max length and mode and none of them did anything

**Reply by JakeStanger** [reply]
Can you share your config please and I'll have a look?

**Reply by HenrySteinmetz** [reply]
```
{
  "monitors": {
    "DP-1": {
      "position": "top",
      "height": 32,
      "start": [
        {
          "type": "workspaces",
          "all_monitors": true,
          "name_map": {
            "1": "●",
            "2": "●",
            "3": "●",
            "4": "●",
            "5": "●",
            "6": "●",
            "7": "●",
            "8": "●",
            "9": "●",
            "10": "●"
          }
        }
      ],
      "center": [
        {
          "type": "music",
          "player_type": "mpris",
          "format": "{title} {elapsed}/{duration}",
          "truncate": "end",
          "icons.play": "",
          "icons.pause": "",
          "icons.prev": "",
          "icons.next": "",
          "truncate.max_length": 100,
          "truncate.mode": "start"
        }
      ],
      "end": [
        {
          "type": "sys_info",
          "interval": 3,
          "format": [
            " {memory_percent}%",
            " {disk_percent:/}%"
          ]
        },

        {
          "type": "clock"
        }
      ],
      "anchor_to_edges": false,
      "margin.left": 0,
      "margin.right": 0
    }
  }
}

```

**Reply by JakeStanger** [reply]
Sorry this slipped off my radar. Is this still an issue? If so I'll make note and try to get to it this week

**Comment by A-Cloud-Ninja** [comment]
Depending on how much content you have filling your bar, I got space on the left and right side of my bar with

```
#bar {
    margin-left: 20px;
    margin-right: 20px;
}

```
